rapid/src/test/java/com/vrg/rapid/integration/AbstractIT.java

```diff
@@ -0,0 +1,241 @@
+                this.inputStream = inputStream;
+                this.logfile = logfile;
+            }
+
+            @Override
+            public void run() {
+                new BufferedReader(new InputStreamReader(inputStream)).lines()
+                        .forEach((x) -> {
+                                    try {
+                                        write(Paths.get(logfile), x.getBytes());
+                                    } catch (Exception e) {
+                                        e.printStackTrace();
+                                    }
+                                }
+                        );
+            }
+        }
+
+        /**
+         * Runs the rapid process with the provided parameters.
+         *
+         * @return RapidNodeRunner
+         * @throws IOException if jar or temp directory is not found.
+         */
+        RapidNodeRunner runNode() throws IOException {
+
+            File rapidRunnerJar = new File(RAPID_RUNNER_JAR);
+            if (!rapidRunnerJar.exists()) {
+                throw new FileNotFoundException("Rapid runner jar not found.");
+            }
+            String command = "java" +
+                    " -server" +
+                    " -jar " + RAPID_RUNNER_JAR +
+                    " --listenAddress " + this.listenAddress +
+                    " --seedAddress " + this.seed +
+                    " --role " + this.role +
+                    " --cluster " + this.clusterName;
+
+            File outputLogFile = new File(OUTPUT_LOG_DIR + File.separator + UUID.randomUUID().toString());
+            System.out.println("Output for listenAddress:" + listenAddress + " logged : " + outputLogFile.getAbsolutePath());
+
+            ProcessBuilder builder = new ProcessBuilder();
+            builder.command("sh", "-c", command);
+            builder.directory(new File(RAPID_PROJECT_DIR));
+            Process rapidProcess = builder.start();
+            OutputLogger outputLogger = new OutputLogger(rapidProcess.getInputStream(), outputLogFile.getAbsolutePath());
+            Executors.newSingleThreadExecutor().submit(outputLogger);
+            this.rapidProcess = rapidProcess;
+            isKilled = false;
+            rapidNodeRunners.add(this);
+
+            return this;
+        }
+
+        private long getPid(Process p) {
+            long pid = -1;
+            try {
+                if (p.getClass().getName().equals("java.lang.UNIXProcess")) {
+                    Field f = p.getClass().getDeclaredField("pid");
+                    f.setAccessible(true);
+                    pid = f.getLong(p);
+                    f.setAccessible(false);
+                }
+            } catch (Exception e) {
+                pid = -1;
+            }
+            return pid;
+        }
+
+        /**
+         * Kills the process on test exit.
+         *
+         * @return
+         */
+        RapidNodeRunner killOnExit() {
```

Comment:
Would there be a case that you would want to use a cluster for a set of tests?
In that case the nodes are not killed.